### PR TITLE
[IMP] mail: fix chatter infinity scroll issue

### PR DESCRIPTION
**Pull request by pka-odoo**
**PURPOSE**

Ease the way users navigate in the chatter.

**SPECIFICATION**

When there are many messages in the chatter, on scrolling down to the bottom,
the next message batch will be automatically loaded(with a spinner) instead of
asking for a "load more.." click.

**LINKS**

TaskId: 2363032

### CODE REVIEW

File: addons/mail/static/src/widgets/form_renderer/form_renderer.js [hunk 1]
```diff
@@ -15,6 +15,9 @@ class ChatterContainerWrapperComponent extends ComponentWrapper {}
  * subset of) the mail widgets (mail_thread, mail_followers and mail_activity).
  */
 FormRenderer.include({
+    events: Object.assign({}, FormRenderer.prototype.events, {
+        'touchmove': '_onScrollForm', // Mobile
+    }),
```
Comment (seb-odoo) [inline addons/mail/static/src/widgets/form_renderer/form_renderer.js:20]: Why? Doesn't a touch event that lead to a scroll change also generate a scroll event?

File: addons/mail/static/src/widgets/form_renderer/form_renderer.js [hunk 2]
```diff
@@ -31,6 +34,13 @@ FormRenderer.include({
         // Do not load chatter in form view dialogs
         this._isFromFormViewDialog = params.isFromFormViewDialog;
     },
+    /**
+     * @override
+     */
+    start() {
+        document.addEventListener('scroll', this._onScrollForm.bind(this), true);
```
Comment (seb-odoo) [inline addons/mail/static/src/widgets/form_renderer/form_renderer.js:41]: There is a syntax in widget to handle event based on class name, that doesn't work for this case?

File: addons/mail/static/src/widgets/form_renderer/form_renderer.js [hunk 3]
```diff
@@ -42,6 +52,12 @@ FormRenderer.include({
         this.off('o_message_posted', this);
         owl.Component.env.bus.off('mail.thread:promptAddFollower-closed', this);
     },
+    /**
+     * @returns {Element|undefined} Scrollable Element
+     */
+    getScrollableElement() {
+        return $(".o_content") && $(".o_content")[0];
```
Comment (seb-odoo) [inline addons/mail/static/src/widgets/form_renderer/form_renderer.js:59]: This condition seems too generous, looking into the whole DOM. Shouldn't it be at least `this.$(` instead?

File: addons/mail/static/src/widgets/form_renderer/form_renderer.js [hunk 4]
```diff
@@ -183,6 +200,17 @@ FormRenderer.include({
...
      */
     _onChatterRendered(ev) {},
+    /**
+     * @private
+     * @param {MouseEvent} ev
+     */
+    _onScrollForm(ev) {
+        const target = ev.target;
+        const chatterContainer = target.querySelector('.o_FormRenderer_chatterContainer');
+        if (this._chatterContainerComponent && chatterContainer && !chatterContainer.classList.contains('o-aside')) {
```
Comment (seb-odoo) [inline addons/mail/static/src/widgets/form_renderer/form_renderer.js:210]: The form rendered itself is already aware if the chatter is aside or not (see `_isChatterAside`), this shouldn't check the DOM.

Note that this makes sense only in enterprise, chatter is never aside in community.

File: addons/mail/static/src/widgets/form_renderer/form_renderer_tests.js [hunk 5]
```diff
@@ -975,6 +975,75 @@ QUnit.test('chatter just contains "creating a new record" message during the cre
     );
 });
 
+QUnit.test('automatic loading of messages on scrolling of formview', async function (assert) {
+    assert.expect(3);
+
+    for (let i=0; i<50; i++) {
```
Comment (seb-odoo) [inline addons/mail/static/src/widgets/form_renderer/form_renderer_tests.js:981]: Use const and apply liter

**Review by seb-odoo — CHANGES_REQUESTED** [review]
Hello, thanks for the PR.

Here is a first check.

File: addons/mail/static/src/widgets/form_renderer/form_renderer.js (hunk 2 above)
Comment (dhp-odoo) [reply addons/mail/static/src/widgets/form_renderer/form_renderer.js:37]: Actually we required scroll event on ".o_content" class. but it is actually a base template. and with event we can not have any custom scroll event based on class.

File: addons/mail/static/src/widgets/form_renderer/form_renderer.js (hunk 3 above)
Comment (dhp-odoo) [reply addons/mail/static/src/widgets/form_renderer/form_renderer.js:55]: We are unable to get the element in this. we have only channelId in this.

File: addons/mail/static/src/widgets/form_renderer/form_renderer_tests.js (hunk 5 above)
Comment (dhp-odoo) [reply addons/mail/static/src/widgets/form_renderer/form_renderer_tests.js:980]: @seb-odoo . not getting you actually.

Comment (seb-odoo) [reply addons/mail/static/src/widgets/form_renderer/form_renderer_tests.js:980]: Forget it for const that was wrong.
But for linter: should have space around operators (=, <)

Comment (dhp-odoo) [reply addons/mail/static/src/widgets/form_renderer/form_renderer_tests.js]: Done

**Comment by C3POdoo** [comment]
Dear @pka-odoo,

Thank you for your contribution but we are closing it due to inactivity.
We only support the last 3 stable versions so no longer accepts patches into this branch.

We apology if we could not look at your request in time.
If the contribution still makes sense, please let us know and do not hesitate to recreate one for the recent versions. We will try to check it as soon as possible.

_This is an automated message._

**Pull request closed without merging**